benchmarks/benchmarks/bench_records.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import absolute_import, division, print_function
+import os
+
+from .common import Benchmark
+
+import numpy as np
+
+
+class Records(Benchmark):
+    def setup(self):
+        self.l50 = np.arange(1000)
+        self.fields_number = 10000
+        self.arrays = [self.l50 for _ in range(self.fields_number)]
+        self.formats = [self.l50.dtype.str for _ in range(self.fields_number)]
+        self.formats_str = ','.join(self.formats)
+        self.dtype_ = np.dtype(
+            [
+                ('field_{}'.format(i), self.l50.dtype.str)
+                for i in range(self.fields_number)
+            ]
+        )
+        self.buffer = self.l50.tostring() * self.fields_number
+        for file in ('test1.bin', 'test2.bin', 'test3.bin'):
+            with open(file, 'w+b') as f:
+                f.write(self.buffer)
```

Comment:
I'm not sure, but I agree that it's adviseable to use a temporsry folder.

Reply:
pytest has a nice fixture for this: just add `tmpdir` as a method argument and then you can do things like `file = str(tmpdir.join('test1.bin'))`

Reply:
Unfortunately this is asv, not pytest

Reply:
Oops, should have looked closer. In that case, we surely have to write to something in memory. Maybe just use `io.StringIO()` instances?

Reply:
`fromfile` won't support that. I think we should just remove the `fromfile`-benchmarks, since it's unlikely they exercise any slow path that we control

Reply:
I've removed the benchmark methods and setup related to ``fromfile``